include/openssl/bio.h

```diff
@@ -724,10 +724,16 @@ void BIO_copy_next_retry(BIO *b);
  * long BIO_ghbn_ctrl(int cmd,int iarg,char *parg);
  */
 
-# ifdef __GNUC__
-#  define __bio_h__attr__ __attribute__
-# else
-#  define __bio_h__attr__(x)
+# define __bio_h__attr__(x)
+# if defined(__GNUC__) && defined(__STDC_VERSION__)
+    /*
+     * Because we support the 'z' modifier, which made its appearance in C99,
+     * we can't use __attribute__ with pre C99 compilers.
```

Comment:
Two things. One is something I kind of know what I'm talking about, and one - not. As for former. In their vast majority, there is no Cxx GNU C compiler. Even 2.95.3 (oldest I have on my computer) can consider code for compliance with multiple standards. I mean comment should really read "pre-C99 *dialect*" or similar. You mentioned that you couldn't reproduce the problem with your compiler. You actually could simply configure with additional -std=c89 option :-) As for latter.  To my programmer's eye/ear sentence's syntax is unparseable. I mean my mind looks for equivalent of `&&` between fact that we support 'z' and its appearance in standard, but doesn't see it, so I don't know if it's there. For this reason I'd prefer if somebody else chimes in :-)